icu4c/source/test/intltest/tztest.cpp

```diff
@@ -2074,6 +2074,8 @@ void TimeZoneTest::TestCanonicalID() {
         {"Asia/Vientiane", "Asia/Bangkok"},
         {"Atlantic/Jan_Mayen", "Europe/Oslo"},
         {"Atlantic/St_Helena", "Africa/Abidjan"},
+        {"Australia/Currie", "Australia/Hobart"},
```

Comment:
This test case is evaluating a set of equivalent zones. The test case was written when canonical ID definition depended on tz database's "Link" rules. The current ICU implementation uses canonical ID mapping coming from CLDR. This old test code is somewhat still useful, but we need to maintain these exceptions. The list of exception mapping has been growing, because tz database tends to use "Link" rule more and more recently (cutting off unreliable differences before 1970).

Reply:
This sounds like it would be useful as a test code comment. Follow-up PR?